### PR TITLE
Restrict what TDML Runner Searches

### PR DESCRIPTION
**Pull request by olabusayoT**
- exclude location information
- update failing tests
- add diagnosticsStripLocationInfo test flag that is set to true as a default, and add as false to tests that need to check the location info
- pass in diagnosticStripFileInfo to verifyAllDiagnosticsFound function

DAFFODIL-541

### CODE REVIEW

**Review by mike-mcgann — APPROVED** [review]

**Review by mbeckerle — APPROVED** [review]
+1
Just need to add some doc strings/comments.

File: daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala [hunk 1]
```diff
@@ -1745,10 +1779,11 @@ object VerifyTestCase {
     actualDiags: Seq[Diagnostic],
     expectedDiags: Option[ErrorWarningBase],
     implString: Option[String],
+    diagnosticStripFileInfo: Boolean = true,
```
[the remark this answers is not in the document]
Comment (mbeckerle) [reply daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala:1782]: I wonder if renaming "diagnosticStripFileInfo" parameter to "avoidFalsePositiveMatchesWithLocationInfo" makes things clearer.

**Review by mbeckerle — APPROVED** [review]
+1

File: daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala [hunk 2]
```diff
@@ -1775,8 +1826,13 @@ object VerifyTestCase {
         expectedDiag.messages.foreach { expectedMsg =>
           {
             val wasFound = actualDiags.exists { actualDiag =>
+              val actualDiagMsg = if (avoidFalsePositiveMatchesWithLocationInfo) {
+                stripLocationInformation(actualDiag.getMessage())
```
[the remark this answers is not in the document]
Comment (mbeckerle) [reply daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala:1830]: Oh yeah, and welcome back!

**Pull request merged**